Sources/DotNetGraph/Edge/DotEdge.cs

```diff
@@ -45,31 +46,43 @@ public DotEdgeArrowHeadAttribute ArrowHead
             get => GetAttribute<DotEdgeArrowHeadAttribute>();
             set => SetAttribute(value);
         }
-        
+
         public DotEdgeArrowTailAttribute ArrowTail
         {
             get => GetAttribute<DotEdgeArrowTailAttribute>();
             set => SetAttribute(value);
         }
-        
+
         public DotPositionAttribute Position
         {
             get => GetAttribute<DotPositionAttribute>();
             set => SetAttribute(value);
         }
-        
+
         public DotEdge()
         {
         }
-        
+
         public DotEdge(IDotElement left, IDotElement right)
         {
-            Left = left;
-            Right = right;
+            Left = left ?? throw new ArgumentNullException(nameof(left));
+            Right = right ?? throw new ArgumentNullException(nameof(right));
         }
 
         public DotEdge(string left, string right)
         {
+            if (left == null)
+                throw new ArgumentNullException(nameof(left));
+
+            if (right == null)
+                throw new ArgumentNullException(nameof(right));
+
+            if (left.Trim().Length == 0)
```

Comment:
You could use method `String.IsNullOrWhiteSpace` instead and maybe merge it with the null check above.